test/fixtures/rules/std_rule_cases/AL09.yml

```diff
@@ -20,79 +20,79 @@ test_pass_no_self_alias_function:
             max(sum) as max_sum
         from foo
 
-test_fail_self_alias:
-    fail_str:
+test_pass_self_alias_sensitivity_change:
+    # Renaming the column name from case-sensitive to case-insensitive
+    pass_str:
         select
-            col_a as col_a,
-            col_b as new_col_b
+            "col_a" as col_a,
```

Comment:
I'm not sure we'd want these to pass - they are the same content, just with quotes. Thoughts?

Reply:
Actually, `"col_a"` is case sensitive, which results in the final column name to be all lower case. However, `col_a` without quotes results in a case-insensitive column name. 

Therefore, the alias here is changing the case sensitivity of the column names. 

Here is an example:

<img width="1241" alt="Screenshot 2024-08-11 at 22 34 07" src="https://github.com/user-attachments/assets/1aa62b2c-8103-4b46-9452-c7af0f287f98">

Reply:
In this example, a CTAS command is used where the case-sensitivity of the column is changed. 

<img width="1248" alt="Screenshot 2024-08-11 at 22 43 45" src="https://github.com/user-attachments/assets/4c37a11c-cd5c-400c-89e9-d8ab5d9b905b">

Reply:
Hi @WittierDinosaur , yes! I had been testing with incorrect dialects all along. This case is more relevant to dialects like Oracle, Snowflake